### PR TITLE
pfEmptyState for Context Views (list, card, table)

### PR DESCRIPTION
**Pull request by dtaylor113**
Angular-Patternfly Implementation of http://www.patternfly.org/pattern-library/communication/empty-state/#/api

pfEmptyState ngdoc:

image

pfEmptyState ngdoc examples for pfListView, pfCardView, pfTableView, and pfToolbar:

image

### CODE REVIEW

File: src/views/cardview/card-view.component.js [hunk 1]
```diff
@@ -88,6 +89,9 @@
            <label class="checkbox-inline">
              <input type="checkbox" ng-model="showDisabled">Show Disabled Cards</input>
            </label>
+           <label class="checkbox-inline">
+ +           <input type="checkbox" ng-model="config.itemsAvailable">Items Available</input>
+ +         </label>
```
Comment (jeff-phillips-18) [inline src/views/cardview/card-view.component.js]: extra +'s in here.

File: src/views/cardview/card-view.component.js [hunk 2]
```diff
@@ -195,12 +211,13 @@
...
   bindings: {
     config: '=?',
+    emptyStateConfig: '=?',
     items: '=',
     eventId: '@id'
   },
   transclude: true,
   templateUrl: 'views/cardview/card-view.html',
-  controller: function (pfUtils) {
+  controller: function (pfUtils, $log) {
```
Comment (jeff-phillips-18) [inline src/views/cardview/card-view.component.js]: is $log used anywhere?

File: src/views/empty-state.component.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+          title: 'Perform an action',
+          actionFn: performAction
+        }
+     ];
+   }
+ ]);
+</file>
+</example>
+*/
+angular.module('patternfly.views').component('pfEmptyState', {
```
Comment (jeff-phillips-18) [inline src/views/empty-state.component.js]: Would prefer example code be in examples directory. Feel free to ignore.

File: src/views/empty-state.component.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    };
+
+    ctrl.$onChanges = function (changesObj) {
+      if ((changesObj.config && !changesObj.config.isFirstChange()) ) {
+        ctrl.updateConfig();
+      }
+    };
+
+    ctrl.areMainActions = function () {
+      var mainActions;
```
Comment (jeff-phillips-18) [inline src/views/empty-state.component.js]: nit: would be clearer if functions were 'hasMainActions' and 'hasSecondaryActions'

File: styles/angular-patternfly.css [hunk 5]
```diff
@@ -512,3 +512,12 @@ table.dataTable tbody th, table.dataTable tbody td {
...
 }
+
+.blank-slate-pf {
+  height:100%;
+  margin-bottom: 0px;
+}
+
+.blank-slate-pf button {
+  margin-right: 4px;
+}
```
Comment (jeff-phillips-18) [inline styles/angular-patternfly.css]: Should these be in patternfly? Maybe just add a comment to explain.

File: src/views/empty-state.component.js (hunk 3 above)
Comment (dtaylor113) [inline src/views/empty-state.component.js]: yeah, typically if there is more than one example I put them in the examples dir.

**Comment by dtaylor113** [comment]
Hi @jeff-phillips-18 , addressed your review issues.
Thanks!

File: README.md [hunk 6]
```diff
@@ -79,14 +79,15 @@ Note:
         <script src="node_modules/angular-patternfly/node_modules/patternfly/node_modules/c3/c3.min.js"></script>
         <script src="node_modules/angular-patternfly/node_modules/patternfly/node_modules/d3/d3.min.js"></script>
 
-5. (optional) The 'patternfly.charts' module is not a dependency in the default angular 'patternfly' module.
-   In order to use patternfly charts you must add 'patternfly.charts' as a dependency in your application:
+5. (optional) The 'patternfly.charts' and 'patternfly.table' modules are not a dependencies in the default angular 'patternfly' module.
```
Comment (dgutride) [inline README.md]: Extra 'a' in this sentence (not a dependencies)

File: README.md [hunk 7]
```diff
@@ -79,14 +79,15 @@ Note:
...
+   In order to use patternfly charts and/or patternfly.table, you must add them as a dependencies in your application:
```
Comment (dgutride) [inline README.md]: Same as above - we should remove the extra 'a'

**Pull request merged**